src/rand.jl

```diff
@@ -0,0 +1,42 @@
+import Random
+
+# TODO: remove Random.CloseOpen01{BigFloat} parameters
+# TODO: remove setprecision
+# possibly fixed by https://github.com/JuliaLang/julia/pull/38169
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{T},
+    n::Random.Repetition,
+) where {T<:Union{Arf,Arb,Acb}}
+    return Random.SamplerSimple(
+        Random.SamplerType{T}(),
+        Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(T)),
+    )
+end
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    x::TOrRef,
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+    T = _nonreftype(TOrRef)
+    return Random.SamplerSimple(
+        Random.SamplerType{T}(),
+        Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(x)),
+    )
+end
+
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arf}) where {A,B} =
```

Comment:
Sorry for not spotting that earlier, but you might want to constrain the `A` type here. For now it probably won't matter, but while Arblib "owns" `Arf`, it doesn't own `SamplerSimple` and doesn't have the exclusivity on using `Arf`. It means that I could depend on Arblib, and write a `Sampler` method which also generates `Arf`, and could therefore write exactly the same method, without realizing I'm overwriting this one (or write another one which introduces ambiguitites). So it's best with `SamplerSimple` to constrain the first type parameter:
```suggestion
Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{Random.SamplerType{Arf},B,Arf}) where {B} =
```
Maybe keeping the trailing `{B,Arf}` as type parameters of `SamplerSimpler` also becomes unnecessary then.

Reply:
thanks for the remark, I changed the methods appropriately